features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/webview/WebViewBrowserSnapshotHandler.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.webview
+
+import android.webkit.WebView
+import com.datadog.android.sessionreplay.internal.utils.RumContextProvider
+import java.util.concurrent.TimeUnit
+
+internal class WebViewBrowserSnapshotHandler(
+    private val rumContextProvider: RumContextProvider,
+    private val dataPurgeTtlLimitInNs: Long = DATA_PURGE_TTL_LIMIT_IN_NS
+) {
+
+    internal val webViewsFullSnapshotState: MutableMap<Int, Pair<String, Long>> = mutableMapOf()
+
+    internal fun triggerFullSnapshotIfNeeded(webView: WebView) {
+        val webViewId = System.identityHashCode(webView)
+        val currentRumViewId = rumContextProvider.getRumContext().viewId
+        val fullSnapshotInfo = webViewsFullSnapshotState[webViewId]
+            ?: Pair(null, System.nanoTime())
+        if (fullSnapshotInfo.first != currentRumViewId) {
+            webViewsFullSnapshotState[webViewId] = Pair(currentRumViewId, System.nanoTime())
+            webView.evaluateJavascript(FORCE_SNAPSHOT_JS_SIGNATURE, null)
+        }
+        purgeWebViewFullSnapshotStateMap()
+    }
+
+    private fun purgeWebViewFullSnapshotStateMap() {
+        val iterator = webViewsFullSnapshotState.entries.iterator()
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        // next/remove can't fail: we checked hasNext and there is not concurrency access
+        // as method is called from the main thread only
```

Comment:
to be fair there is no `MainThread` annotation here and in the call chain as well, maybe add it?